speech_to_text.py

```diff
@@ -114,6 +114,13 @@ def run_whisper(job):
         writer(result, media_file, writer_options)
 
     job["finished"] = now()
+    tech_md_dict = {
+        "speech_to_text_extraction_program": {"name": "whisper", "version": whisper.version.__version__},
+        "effective_options": options,
+        "effective_writer_options": writer_options,
+        "effective_model_name": model_name
```

Comment:
another thing i considered adding, but left off as possibly too verbose and not useful enough: `"effective_model": str(model)`

this produced output like:
```
Whisper(\n  (encoder): AudioEncoder(\n    (conv1): Conv1d(80, 768, kernel_size=(3,), stride=(1,), padding=(1,))\n    (conv2): Conv1d(768, 768, kernel_size=(3,), stride=(2,), padding=(1,))\n    (blocks): ModuleList(\n      (0-11): 12 x ResidualAttentionBlock(\n        (attn): MultiHeadAttention(\n          (query): Linear(in_features=768, out_features=768, bias=True)\n          (key): Linear(in_features=768, out_features=768, bias=False)\n          (value): Linear(in_features=768, out_features=768, bias=True)\n          (out): Linear(in_features=768, out_features=768, bias=True)\n        )\n        (attn_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)\n        (mlp): Sequential(\n          (0): Linear(in_features=768, out_features=3072, bias=True)\n          (1): GELU(approximate='none')\n          (2): Linear(in_features=3072, out_features=768, bias=True)\n        )\n        (mlp_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)\n      )\n    )\n    (ln_post): LayerNorm((768,), eps=1e-05, elementwise_affine=True)\n  )\n  (decoder): TextDecoder(\n    (token_embedding): Embedding(51865, 768)\n    (blocks): ModuleList(\n      (0-11): 12 x ResidualAttentionBlock(\n        (attn): MultiHeadAttention(\n          (query): Linear(in_features=768, out_features=768, bias=True)\n          (key): Linear(in_features=768, out_features=768, bias=False)\n          (value): Linear(in_features=768, out_features=768, bias=True)\n          (out): Linear(in_features=768, out_features=768, bias=True)\n        )\n        (attn_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)\n        (cross_attn): MultiHeadAttention(\n          (query): Linear(in_features=768, out_features=768, bias=True)\n          (key): Linear(in_features=768, out_features=768, bias=False)\n          (value): Linear(in_features=768, out_features=768, bias=True)\n          (out): Linear(in_features=768, out_features=768, bias=True)\n        )\n        (cross_attn_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)\n        (mlp): Sequential(\n          (0): Linear(in_features=768, out_features=3072, bias=True)\n          (1): GELU(approximate='none')\n          (2): Linear(in_features=3072, out_features=768, bias=True)\n        )\n        (mlp_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)\n      )\n    )\n    (ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)\n  )\n)
```

the non-stringified and slightly more readable version i got when playing around with this in parts on the python REPL):
```
>>> model
Whisper(
  (encoder): AudioEncoder(
    (conv1): Conv1d(80, 768, kernel_size=(3,), stride=(1,), padding=(1,))
    (conv2): Conv1d(768, 768, kernel_size=(3,), stride=(2,), padding=(1,))
    (blocks): ModuleList(
      (0-11): 12 x ResidualAttentionBlock(
        (attn): MultiHeadAttention(
          (query): Linear(in_features=768, out_features=768, bias=True)
          (key): Linear(in_features=768, out_features=768, bias=False)
          (value): Linear(in_features=768, out_features=768, bias=True)
          (out): Linear(in_features=768, out_features=768, bias=True)
        )
        (attn_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
        (mlp): Sequential(
          (0): Linear(in_features=768, out_features=3072, bias=True)
          (1): GELU(approximate='none')
          (2): Linear(in_features=3072, out_features=768, bias=True)
        )
        (mlp_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
      )
    )
    (ln_post): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
  )
  (decoder): TextDecoder(
    (token_embedding): Embedding(51865, 768)
    (blocks): ModuleList(
      (0-11): 12 x ResidualAttentionBlock(
        (attn): MultiHeadAttention(
          (query): Linear(in_features=768, out_features=768, bias=True)
          (key): Linear(in_features=768, out_features=768, bias=False)
          (value): Linear(in_features=768, out_features=768, bias=True)
          (out): Linear(in_features=768, out_features=768, bias=True)
        )
        (attn_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
        (cross_attn): MultiHeadAttention(
          (query): Linear(in_features=768, out_features=768, bias=True)
          (key): Linear(in_features=768, out_features=768, bias=False)
          (value): Linear(in_features=768, out_features=768, bias=True)
          (out): Linear(in_features=768, out_features=768, bias=True)
        )
        (cross_attn_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
        (mlp): Sequential(
          (0): Linear(in_features=768, out_features=3072, bias=True)
          (1): GELU(approximate='none')
          (2): Linear(in_features=3072, out_features=768, bias=True)
        )
        (mlp_ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
      )
    )
    (ln): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
  )
)
```

side-note looks oddly LISP-y to me (says someone who last wrote LISP in a college class 20+ years ago.

Reply:
Possibly worth a post-standup discussion. I would not find this level of detail useful to record, but perhaps someone more familiar with whisper would see some data in there they'd want?